### ISSUE TITLE
isVideo not properly set for some files

### ISSUE THREAD

**Issue by s-ol** [VeliovGroup/Meteor-Files#361]
I don't really understand this bug from looking at the source, but i just confirmed it on [files.veliov.com](https://files.veliov.com/) and the `demo-simplest-upload`:  
`isVideo` is not properly set with some video types, specifically matroska videos (`mkv`, at least thats how I noticed it).

You can download my test file from your demo (https://files.veliov.com/QCsG25yCS3eaKARuo) or from the page I originally got it: [mkv at jell.yfish.us](http://jell.yfish.us/media/jellyfish-3-mbps-hd-h264.mkv)

If I log fileRef in `onAfterUpload` I see

    isVideo: false,
    type: 'video/x-matroska',

which should satisfy [the check in line 1242](https://github.com/VeliovGroup/Meteor-Files/blob/master/files.coffee#L1242):

```coffee
      isVideo: /^video\//i.test data.type
      isAudio: /^audio\//i.test data.type
      isImage: /^image\//i.test data.type
      isText:  /^text\//i.test data.type
      isJSON:  /application\/json/i.test data.type
      isPDF:   /application\/pdf|application\/x-pdf/i.test data.type
```

Or is the mimeType that `isVideo` is checked against the one sent by the client, which is later replaced by a `file-type` ([_getMimeType()](https://github.com/VeliovGroup/Meteor-Files/blob/master/files.coffee#L1139))?
If so, shouldn't the `is*` be updated at that time, too?

---

also by the way, why not

```coffee
      isVideo: data.type.startsWith('video/')
      isAudio: data.type.startsWith('audio/')
      isImage: data.type.startsWith('image/')
      isText:  data.type.startsWith('text/')
      isJSON:  data.type === 'application/json'
      isPDF:   data.type === 'application/pdf' || data.type === 'application/x-pdf'
// or at least
      isPDF:   /^application\/(-x)?pdf$/i.test data.type
```

I would assume that to be more performant and I think it reads better.

**Comment by s-ol** [comment]
@dr-dimitru ah, I didn't know about IE.
I'm not sure how blocking/threatening this is to us, but I might work on this. If my hypothesis is correct then it shouldn't be a large change.

**Comment by dr-dimitru** [comment]
PR is merged and package is published as [`v1.7.12`](https://github.com/VeliovGroup/Meteor-Files/releases/tag/1.7.12), thank you for contribution
